hoomd/hpmc/compute.py

```diff
@@ -7,123 +7,111 @@
+                                _hoomd.CellList(self._simulation.state._cpp_sys_def),
+                                "")
+
+        super()._attach()
+
+    @log
+    def free_volume(self):
+        """free volume available to a particle assembly
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.free_volume
         else:
-            if isinstance(mc, integrate.sphere):
-                cls = _hpmc.ComputeFreeVolumeGPUSphere;
-            elif isinstance(mc, integrate.convex_polygon):
-                cls = _hpmc.ComputeFreeVolumeGPUConvexPolygon;
-            elif isinstance(mc, integrate.simple_polygon):
-                cls = _hpmc.ComputeFreeVolumeGPUSimplePolygon;
-            elif isinstance(mc, integrate.convex_polyhedron):
-                cls = _hpmc.ComputeFreeVolumeGPUConvexPolyhedron;
-            elif isinstance(mc, integrate.convex_spheropolyhedron):
-                cls = _hpmc.ComputeFreeVolumeGPUSpheropolyhedron;
-            elif isinstance(mc, integrate.ellipsoid):
-                cls = _hpmc.ComputeFreeVolumeGPUEllipsoid;
-            elif isinstance(mc, integrate.convex_spheropolygon):
-                cls =_hpmc.ComputeFreeVolumeGPUSpheropolygon;
-            elif isinstance(mc, integrate.faceted_ellipsoid):
-                cls =_hpmc.ComputeFreeVolumeGPUFacetedEllipsoid;
-            elif isinstance(mc, integrate.polyhedron):
-                cls =_hpmc.ComputeFreeVolumeGPUPolyhedron;
-            elif isinstance(mc, integrate.sphinx):
-                cls =_hpmc.ComputeFreeVolumeGPUSphinx;
-            elif isinstance(mc, integrate.sphere_union):
-                cls = _hpmc.ComputeFreeVolumeGPUSphereUnion;
-            elif isinstance(mc, integrate.faceted_ellipsoid_union):
-                cls = _hpmc.ComputeFreeVolumeGPUFacetedEllipsoidUnion;
-            elif isinstance(mc, integrate.convex_spheropolyhedron_union):
-                cls = _hpmc.ComputeFreeVolumeGPUConvexPolyhedronUnion;
-            else:
-                hoomd.context.current.device.cpp_msg.error("compute.free_volume: Unsupported integrator.\n");
-                raise RuntimeError("Error initializing compute.free_volume");
-
-        if suffix != '':
-            suffix = '_' + suffix
-
-        self.cpp_compute = cls(hoomd.context.current.system_definition,
-                                mc.cpp_integrator,
-                                cl,
-                                seed,
-                                suffix)
-
-        if test_type is not None:
-            itype = hoomd.context.current.system_definition.getParticleData().getTypeByName(test_type)
-            self.cpp_compute.setTestParticleType(itype)
-        if nsample is not None:
-            self.cpp_compute.setNumSamples(int(nsample))
-
-        hoomd.context.current.system.addCompute(self.cpp_compute, self.compute_name)
-        self.enabled = True
+            return None
```

Comment:
@bdice I recall you some alternate to returning `None` in these cases. What was that suggestion?